python-package/xgboost/sklearn.py

```diff
@@ -212,6 +212,105 @@ def adddoc(cls):
     return adddoc
 
 
+def _wrap_evaluation_matrices(
+    missing: float,
+    X: Any,
+    y: Any,
+    group: Optional[Any],
+    qid: Optional[Any],
+    sample_weight: Optional[Any],
+    base_margin: Optional[Any],
+    feature_weights: Optional[Any],
+    eval_set: Optional[List[Tuple[Any, Any]]],
+    sample_weight_eval_set: Optional[List[Any]],
+    base_margin_eval_set: Optional[List[Any]],
+    eval_group: Optional[List[Any]],
+    eval_qid: Optional[List[Any]],
+    create_dmatrix: Callable,
+    label_transform: Callable = lambda x: x,
+) -> Tuple[Any, Optional[List[Tuple[Any, str]]]]:
```

Comment:
```suggestion
) -> Tuple[EvalMatrix, Optional[List[Tuple[EvalMatrix, str]]]]:
```
Let's define a union type `EvalMatrix` as `Union[DMatrix, "DaskDMatrix"]`. This way, we can specifically annotate the return type of `_wrap_evaluation_matrices`.

Reply:
@hcho3 We can't, the `DaskDMatrix` is not defined on this module.  I tried using `TypeVar` but there seems to be failure in type deduction.